src/extension.ts

```diff
@@ -6,44 +6,78 @@ import { removeReference } from './dotnet/remove/removeReference';
 
 export function activate(context: vscode.ExtensionContext) {
-	
-	context.subscriptions.push(vscode.commands.registerCommand('extension.dotnet.add.package', 
-		async () => runAndshowErrorAsMessage(addPackage)));
-
-	context.subscriptions.push(vscode.commands.registerCommand('extension.dotnet.remove.package', 
-		async () => runAndshowErrorAsMessage(removePackage)));
-	
-	context.subscriptions.push(vscode.commands.registerCommand('extension.dotnet.add.reference', 
-		async () => runAndshowErrorAsMessage(addReference)));
-
-	context.subscriptions.push(vscode.commands.registerCommand('extension.dotnet.remove.reference', 
-		async () => runAndshowErrorAsMessage(removeReference)));
-	
-	context.subscriptions.push(vscode.commands.registerCommand('extension.dotnet.sln.add', 
-		async () => runAndshowErrorAsMessage(addProject)));
-
-	context.subscriptions.push(vscode.commands.registerCommand('extension.dotnet.sln.remove', 
-		async () => runAndshowErrorAsMessage(removeProject)));
-
-	context.subscriptions.push(vscode.commands.registerCommand('extension.dotnet.new', 
-		async () => runAndshowErrorAsMessage(newFromTemplate)));
+    outputChannel = vscode.window.createOutputChannel("dotnet ef");
+    context.subscriptions.push(outputChannel);
+
+    context.subscriptions.push(vscode.commands.registerCommand('extension.dotnet.add.package',
+        async () => runAndshowErrorAsMessage(addPackage)));
+    context.subscriptions.push(vscode.commands.registerCommand('extension.dotnet.remove.package',
+        async () => runAndshowErrorAsMessage(removePackage)));
+    context.subscriptions.push(vscode.commands.registerCommand('extension.dotnet.add.reference',
+        async () => runAndshowErrorAsMessage(addReference)));
+    context.subscriptions.push(vscode.commands.registerCommand('extension.dotnet.remove.reference',
+        async () => runAndshowErrorAsMessage(removeReference)));
+    context.subscriptions.push(vscode.commands.registerCommand('extension.dotnet.sln.add',
+        async () => runAndshowErrorAsMessage(addProject)));
+    context.subscriptions.push(vscode.commands.registerCommand('extension.dotnet.sln.remove',
+        async () => runAndshowErrorAsMessage(removeProject)));
+    context.subscriptions.push(vscode.commands.registerCommand('extension.dotnet.new',
+        async () => runAndshowErrorAsMessage(newFromTemplate)));
+    context.subscriptions.push(vscode.commands.registerCommand('extension.dotnet.ef.migrations.add',
+        async () => runAndshowErrorAsMessage(addMigration)));
+    context.subscriptions.push(vscode.commands.registerCommand('extension.dotnet.ef.migrations.remove',
+        async () => runAndshowErrorAsMessage(removeMigration, true)));
+    context.subscriptions.push(vscode.commands.registerCommand('extension.dotnet.ef.database.update',
+        async () => runAndshowErrorAsMessage(updateDatabase, true)));
+    context.subscriptions.push(vscode.commands.registerCommand('extension.dotnet.ef.dbcontext.info',
```

Comment:
I am not sure about the `list` and `info` commands. I do not have them currently for `dotnet`, mainly because they only print output. I want to think a little bit about that, but do you really use both and when?